File: test/language/subtypes/python/UInt16SubtypeTest.py
import Subtypes


class UInt16SubtypeTest(Subtypes.TestCase):
    def testSubtype(self):
        identifier = self.api.Identifier(0xFFFF)
        testStructure = self.api.TestStructure()
        testStructure.identifier = identifier
        readIdentifier = testStructure.identifier
        self.assertEqual(identifier, readIdentifier)

File: test/language/subtypes/cpp/UInt16SubtypeTest.cpp
#include <cstdio>
#include <string>
#include <fstream>

#include "gtest/gtest.h"

#include "subtypes/uint16_subtype/TestStructure.h"

namespace subtypes
{
namespace uint16_subtype
{

TEST(UInt16SubtypeTest, testSubtype)
{
    const Identifier identifier = 0xFFFF;
    TestStructure testStructure;
    testStructure.setIdentifier(identifier);
    const Identifier readIdentifier = testStructure.getIdentifier();
    ASSERT_EQ(identifier, readIdentifier);
}

} // namespace uint16_subtype
} // namespace subtypes
